### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by lwjglgamedev** [lwjglgamedev/vulkanbook#23] [Ideas]
Althogh it seems that the audience of this book seems to be quite narrow, I would like to ask about your opinion on future contents. I'm thinking on try to explain a bindless approach and later on to explot multi-threading. Any thoughs on this? Any other idea?

**Comment by SWinxy** [comment]
Multithreading would be a good idea. There could also be compute pipelines or compute shaders included. There could also be a chapter for [LWJGX/lwjgl3-awt](https://github.com/LWJGLX/lwjgl3-awt).

**Reply by lwjglgamedev** [reply]
Thanks, Just for you to know. Compute shaders are already included in the animation chapters. However, the path tha I'm thinkg is to go to a bindless design, apllying CPU culling with compute shaders and after that, sepparate logic and renderiung in different threads.

I'm also, as  a side project, working in supporting KTX textures.

**Comment by tlf30** [comment]
Another advanced topic that I think people would be interested in is basic raytracing since that is now part of the official spec.

**Comment by lwjglgamedev** [comment]
Yes this is a high-interest topic. The issue that I have is that my GPU does not support raytracing. I have an GeForce GTX 1650 Ti and as I far as I know raytracing extensions are not supported. Therefore, I'm blocked on this at this moment.

**Reply by tlf30** [reply]
I just put in my GTX 1080 TI for some testing and noticed that both `VK_KHR_ray_tracing_pipeline` and `VK_NV_ray_tracing` where listed under device extensions. I have not tested any RTX code on it, but was surprised to see the extensions present. You may check on your GPU to see if they are present as well.

**Reply by tlf30** [reply]
Just tested my GTX 1080 TI using https://github.com/GPSnoopy/RayTracingInVulkan. I was getting a whopping 16 fps, but it still worked. I imagine that your GTX 1650 TI should work with RTX just fine.

**Reply by lwjglgamedev** [reply]
Mmm it sems that my 1650 TI do not have ray tracing extensions.... The example fails

**Reply by tlf30** [reply]
Hmm, that is odd. From what I have read it should support it the same as the gtx 1080 ti. Are you running the latest drivers?

**Reply by lwjglgamedev** [reply]
Yes, I tested also with my own code, and the extensions do not appear. I'm using it on a laptop.
